### PR TITLE
PHP Snippet generation

### PR DESCRIPTION
**Pull request by SilasKenneth**
## Overview

Add snippet generation support for PHP Kiota based SDK.

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

### CODE REVIEW

File: pipelines/snippets.yml [hunk 1]
```diff
@@ -44,7 +45,7 @@ steps:
 
 - pwsh: |
     # override build languag incase provided in pipeline
-    $buildLanguages = if ("$env:LANGUAGES") { "$env:LANGUAGES" } else { "$(snippetLanguages)" }
```
Comment (SilasKenneth) [inline pipelines/snippets.yml:47]: For testing purposes.

**Review by zengin — CHANGES_REQUESTED** [review]
@rkodev could PHP also use the `SnippetCodeGraph` you have created for OpenAPI based generation?

File: pipelines/snippets.yml [hunk 2]
```diff
@@ -44,7 +45,7 @@ steps:
...
+    $buildLanguages = if ("$env:LANGUAGES") { "$(snippetLanguage)" } else { "$(snippetLanguages)" }
```
Comment (rkodev) [inline pipelines/snippets.yml:48]: You can use create the pipeline varaible `LANGUAGE` and pass `php` to the pipeline. Commiting PHP to the pipeline will force any test pipeline to create a new branch / commit to test

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    {
+        var indentManager = new IndentManager();
+        var snippetBuilder = new StringBuilder(
+            "//THIS SNIPPET IS A PREVIEW FOR THE KIOTA BASED SDK. NON-PRODUCTION USE ONLY" + Environment.NewLine +
+            $"{clientVarName} = new {clientVarType}({httpCoreVarName});{Environment.NewLine}{Environment.NewLine}");
+        var (requestPayload, payloadVarName) = GetRequestPayloadAndVariableName(snippetModel, indentManager);
+        snippetBuilder.Append(requestPayload);
+        var responseAssignment = "$result = ";
+        // have a return type if we have a response schema that is not an error
+        if (snippetModel.ResponseSchema == null || (snippetModel.ResponseSchema.Properties.Count == 1 && snippetModel.ResponseSchema.Properties.First().Key.Equals("error",StringComparison.OrdinalIgnoreCase)))
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs:28]:
```suggestion
        if (snippetModel.ResponseSchema == null || (snippetModel.ResponseSchema.Properties.Any() && snippetModel.ResponseSchema.Properties.First(x => x.Key.Equals("error",StringComparison.OrdinalIgnoreCase)))
```

**Review by baywet — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+        var bodyParameter = codeGraph.HasBody()
+            ? $"{RequestBodyVarName}"
+            : string.Empty;
+        var optionsParameter = codeGraph.HasOptions() ? "options" : string.Empty;
+        var returnVar = codeGraph.HasReturnedBody() ? "$requestResult = " : string.Empty;
+        var parameterList = GetActionParametersList(bodyParameter, configParameter, optionsParameter);
+        payloadSb.AppendLine(GetRequestConfiguration(codeGraph, indentManager));
+        payloadSb.AppendLine($"{returnVar}{clientVarName}->{GetFluentApiPath(codeGraph.Nodes)}->{method}({parameterList});");
+    }
+    private const string QueryParametersVarName = "queryParameters";
```
Comment (rkodev) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs:47]: nit: declare constants at the start of the file

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs [hunk 5]
```diff
@@ -107,7 +107,7 @@ private static List<CodeProperty> parseParameters(SnippetModel snippetModel)
         {
 
             var ArrayParameters = ImmutableHashSet.Create("select", "expand", "orderby");
-            var NumberParameters = ImmutableHashSet.Create("skip");
+            var NumberParameters = ImmutableHashSet.Create("skip", "top");
```
Comment (rkodev) [inline CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:110]: nice!

**Review by calebkiage — APPROVED** [review]

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+                payloadSb.Append($"{builder.ToString()},");
+        }
+
+        indentManager.Unindent();
+    }
+
+    private static void WriteEnumValue(StringBuilder payloadSb,string parentPropertyName, CodeProperty currentProperty)
+    {
+        var enumParts = currentProperty.Value.Split('.');
+        var enumClass = enumParts.First();
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs:285]: wrap with if any condition to avoid failure?

**Review by baywet — APPROVED** [review]

**Review by zengin — APPROVED** [review]

**Pull request merged**